replication-api-impl/src/test/java/com/connexta/replication/api/impl/persistence/pojo/FilterIndexPojoTest.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.persistence.pojo;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import com.jparams.verifier.tostring.NameStyle;
+import com.jparams.verifier.tostring.ToStringVerifier;
+import java.time.Instant;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FilterIndexPojoTest {
+
+  private static final Instant MODIFIED_SINCE = Instant.now();
+
+  private static final String ID = "id";
+
+  private static final String FILTER_ID = "filterId";
+
+  private static final FilterIndexPojo POJO =
+      new FilterIndexPojo().setId(ID).setModifiedSince(MODIFIED_SINCE).setFilterId(FILTER_ID);
+
+  private FilterIndexPojo pojo;
+
+  @Before
+  public void setup() {
+    pojo = new FilterIndexPojo();
+  }
+
+  @Test
+  public void testEqualsReflexive() {
+    assertThat(POJO.equals(POJO), is(true));
+  }
+
+  @Test
+  public void testEqualsSymmetric() {
+    final FilterIndexPojo pojo2 =
+        new FilterIndexPojo().setId(ID).setModifiedSince(MODIFIED_SINCE).setFilterId(FILTER_ID);
+    assertThat(POJO.equals(pojo2), is(true));
+    assertThat(pojo2.equals(POJO), is(true));
+  }
+
+  @Test
+  public void testEqualsTransitive() {
+    final FilterIndexPojo pojo2 =
+        new FilterIndexPojo().setId(ID).setModifiedSince(MODIFIED_SINCE).setFilterId(FILTER_ID);
+    final FilterIndexPojo pojo3 =
+        new FilterIndexPojo().setId(ID).setModifiedSince(MODIFIED_SINCE).setFilterId(FILTER_ID);
+    assertThat(POJO.equals(pojo2), is(true));
+    assertThat(pojo2.equals(pojo3), is(true));
+    assertThat(POJO.equals(pojo3), is(true));
+  }
+
+  @Test
+  public void testEqualsConsistent() {
+    final FilterIndexPojo pojo2 =
+        new FilterIndexPojo().setId(ID).setModifiedSince(MODIFIED_SINCE).setFilterId(FILTER_ID);
+    assertThat(POJO.equals(pojo2), is(true));
+    assertThat(POJO.equals(pojo2), is(true));
+  }
+
+  @Test
+  public void testEqualsNull() {
+    assertThat(POJO.equals(null), is(false));
+  }
```

Comment:
❗️ Missing test cases that verify when passing a different class object and also when the various fields are different.

Reply:
❗️ Still missing a test case for equals() when an object that is not a filter index is passed in